hyperspy/component.py

```diff
@@ -1013,10 +1009,7 @@ def __call__(self):
         -------
         numpy array
         """
-
-        axis = self.model.axis.axis[self.model.channel_switches]
-        component_array = self.function(axis)
-        return component_array
+        return self.model.__call__(component_list=[self])
```

Comment:
I think that this is a dangerous change:

*  ``__call__`` should always return the unconvolved component as it was doing previously as the output should not depend on the model and its particular ``convolve`` setting.
* ``__call__`` should work also when ``self.model`` is not defined.

Reply:
Yes, reverted to previous version and move the `model.__call__()` to `_component2plot`.